### PR TITLE
Persistence

### PR DESCRIPTION
**Pull request by sddhrthrt**
Added database layer, persistence in Mysql.

- [x] new found availabilities by poller is stored in `Pincode.availabilities`.
- [x] `Subscription.pincodes` should give each subscription's pincodes.
- [x] poller should be able to query subscriptions for whom new availabilities were fetched, and be able to call `notify_email` with full details of availabilities. `notify_email` not implemented yet.
- [ ] detect duplicate notification, and don't send 

Todo after this:

- `notify_email` implementation, maybe using Kiran's 
- Write APIs for creating new subscription
- Add auth layer, usernames, passwords
- batch notifications together and send once in half hour option?
- Profit

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return f'Pin<{self.code}>'
+
+
+class Subscription(db.Model):
+    __tablename__ = 'subscriptions'
+    '''Class object to represent every new _request_ added to the system by users'''
+    id = db.Column(db.Integer, primary_key=True)
+    old = db.Column(db.Boolean)
+    want_free = db.Column(db.Boolean)
+    flavor = db.Column(db.String(32))
```
Comment (nikhil-soraba) [inline models.py:40]: These two are optional (basically we need another bit for signifying "don't care")

File: sessions.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+def hash_session(session_response):
+    # Here we'll use all "significant" fields of sessions
+    # and form a tuple out of them. A tuple can then be hashed
+    # to give a unique hash to this session.
+    # Goal is for hash to change if any field that we care about
+    # in a session changes. 
+    data = \
+        tuple((session_response.get(s, None) for s in (
+            "center_id", "from", "to", "fee_type", "session_id",
+            "date", "available_capacity", "min_age_limit", "vaccine"
```
Comment (nikhil-soraba) [inline sessions.py]: Umm, so for `availability`, do we really care if it changed from `N` to `N-x` as long as `availability>0`?

File: models.py (hunk 1 above)
Comment (sddhrthrt) [reply models.py:40]: Hmm. None is don't care, no?

File: sessions.py (hunk 2 above)
Comment (sddhrthrt) [reply sessions.py]: ah you're right - this shouldn't matter. Will change that.

File: models.py (hunk 1 above)
Comment (nikhil-soraba) [reply models.py:40]: How does `db` store None? Can a boolean variable in `db` be set to None?

**Pull request merged**